SEFramework/src/lib/CoordinateSystem/WCS.cpp

```diff
@@ -61,9 +61,31 @@ WCS::WCS(const FitsImageSource& fits_image_source) : m_wcs(nullptr, nullptr) {
   int number_of_records = 0;
   auto fits_headers = fits_image_source.getFitsHeaders(number_of_records);
 
+  init(&(*fits_headers)[0], number_of_records);
+}
+
+WCS::WCS(const WCS& original) : m_wcs(nullptr, nullptr) {
+
+  //FIXME Horrible hack: I couldn't figure out how to properly do a deep copy wcsprm so instead
+  // of making a copy, I use the ascii headers output from the original to recreate a new one
```

Comment:
[`wcssub`](https://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/wcs_8h.html#a864c99fef9f3eee29085ce42d0ee0d64) ?

Reply:
Yes, I read the documentation, thank you. As you can guess, I went with this workaround because it didn't work.

Now I'm sure this is because I didn't fully understand the idiosyncrasies of wcslib but I lost a lot of time trying to so I had to find a pragmatic solution instead.

Reply:
That's unfortunate. It will have to be like this, then 🤷🏼